src/react_native/fast_image.cljs

```diff
@@ -1,40 +1,71 @@
+    (string/replace-first (oops/oget source :uri) #":\d+" "")))
 
-(defn fast-image
+(defn- placeholder
+  [{:keys [style fallback-content error? loaded?]}]
+  [rn/view
+   {:style (assoc style
+                  :flex            1
+                  :justify-content :center
+                  :align-items     :center)}
+   (cond
+     (and error? fallback-content) fallback-content
+     error?                        [rn/text "X"]
+     (not loaded?)                 [rn/activity-indicator {:animating true}])])
+
+;; We cannot use hooks since `reactify-component` seems to ignore the functional compiler
+(defn- internal-fast-image
   [_]
-  (let [loaded? (reagent/atom false)
-        error?  (reagent/atom false)]
-    (fn [{:keys [source fallback-content] :as props}]
-      [fast-image-class
-       (merge
-        props
-        {:source   (if (string? source)
-                     {:uri      source
-                      :priority :high}
-                     source)
-         :on-error (fn [e]
-                     (when-let [on-error (:on-error props)]
-                       (on-error e))
-                     (reset! error? true))
-         :on-load  (fn [e]
-                     (when-let [on-load (:on-load props)]
-                       (on-load e))
-                     (reset! loaded? true)
-                     (reset! error? false))})
+  (let [loaded?         (reagent/atom false)
+        error?          (reagent/atom false)
+        on-image-error  (fn [event on-error]
+                          (when (fn? on-error) (on-error event))
+                          (reset! error? true))
+        on-image-loaded (fn [event on-load]
+                          (when (fn? on-load) (on-load event))
+                          (reset! loaded? true)
+                          (reset! error? false))]
+    (fn [{:keys [source fallback-content on-error on-load] :as props}]
+      [:> FastImage
+       (assoc props
+              :source   (build-source source)
+              :on-error #(on-image-error % on-error)
+              :on-load  #(on-image-loaded % on-load))
        (when (or @error? (not @loaded?))
-         [placeholder (:style props)
-          (if @error?
-            (or fallback-content [rn/text "X"])
-            (when-not @loaded?
-              [rn/activity-indicator {:animating true}]))])])))
+         [placeholder
+          {:style            (js->clj (:style props))
+           :fallback-content fallback-content
+           :error?           @error?
+           :loaded?          @loaded?}])])))
+
+(defn- compare-sources
+  [old-props new-props]
+  (let [old-source (oops/oget old-props :source)
+        new-source (oops/oget new-props :source)]
+    (and old-source
+         new-source
+         (= (remove-port old-source) (remove-port new-source)))))
+
+(def fast-image
+  (-> internal-fast-image
+      (reagent/reactify-component)
+      (rn/memo compare-sources)
+      (reagent/adapt-react-class)))
```

Comment:
@ilmotta took your suggestion here, it actually fixes the problem. Thanks!

I prefer this solution since we aren't duplicating a component. mantainability isn't hurt since `reactify-component` was used.

BTW, for some reason, reactify-component seems to ignore the functional compiler, so no hooks were used (the comment on the code has been already added).

Reply:
The way you used `reactify-component` made things much cleaner 👍🏼